db.go

```diff
@@ -602,7 +602,7 @@ func (db *DB) writeRequests(reqs []*request) error {
 		}
 		count += len(b.Entries)
 		var i uint64
-		for err := db.ensureRoomForWrite(); err == errNoRoom; err = db.ensureRoomForWrite() {
+		for err = db.ensureRoomForWrite(); err == errNoRoom; err = db.ensureRoomForWrite() {
```

Comment:
Is `go vet` okay with this? We are still shadowing the previous value. Wouldn't it be better if we change the lines above to -

```
if err := db.vlog.write(reqs); err != nil {
	done(err)
	return err
}
```

and then, create the variable inside the loop?

Reply:
Seems like we are using this variable outside the for loop. In that case, we should declare it before the loop and then use it?

Reply:
err was already defined above on line 591 of this file.

Reply:
:+1: